R/read_redcap.R

```diff
@@ -72,20 +73,83 @@ read_redcap <- function(redcap_uri,
                         forms = NULL,
                         export_survey_fields = TRUE,
                         suppress_redcapr_messages = TRUE) {
-
   check_arg_is_character(redcap_uri, len = 1, any.missing = FALSE)
+  check_arg_is_character(token, len = 1, any.missing = FALSE)
   check_arg_is_valid_token(token)
   check_arg_choices(raw_or_label, choices = c("label", "raw"))
   check_arg_is_character(forms, min.len = 1, null.ok = TRUE, any.missing = FALSE)
   check_arg_is_logical(export_survey_fields, len = 1, any.missing = FALSE)
   check_arg_is_logical(suppress_redcapr_messages, len = 1, any.missing = FALSE)
 
   # Load REDCap Metadata ----
-  db_metadata <- redcap_metadata_read(
-    redcap_uri = redcap_uri,
-    token = token,
-    verbose = !suppress_redcapr_messages
-  )$data %>%
+  # Capture unexpected metadata API call errors
+  try_fetch(
+    {
+      call <- current_call()
+
+      out <- redcap_metadata_read(
```

Comment:
I think this would be more consistent with the flow of `read_redcap` since we introduce an `out` object on L220
```suggestion
      db_metadata <- redcap_metadata_read(
```

Reply:
Done!